### PR TITLE
Use hdf5 1.12.1 on all platforms to have consistent dependencies

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Use hdf5 1.12.1 on all platforms to have consistent dependencies.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,25 +14,24 @@ source:
...
 requirements:
   build:
     - {{ compiler("c") }}
   host:
     - python
-    - cython >=0.29.15
-    - hdf5
-    - numpy
+    - cython >=0.29.15,<3
+    - hdf5 1.12.1
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:26]: if the goal is to use hdf5 1.12.1 everywhere would not be better to add hdf to aggregate/conda_build_config.yaml to separate aggregate branch ?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:26]: I sincerely think that CBC would be bad here. It's there right now, but there is two versions (one for osx-arm64 and one for the others) and I'll sincerely say that all this worked was caused by CBC having different versions.

Generally, when I speak about consistency, I mean consistency between variants/platforms for a version X. For example, if you use h5py on mac and you get hdf5-1.12.1 and on linux you get 1.10.1, this is inconsistent and quite problematic. When I use a version of a package, I expect it to have the same set of dependencies across platforms. Even more so when there is a file format (or similar things like hdf5) involved.

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:26]: yes, definitely there are pros and cons, so if you think this is better then OK with me

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by cbouss — COMMENTED** [review]
This looks good to me, but could you update to h5py 3.9.0?
https://anaconda.atlassian.net/browse/PKG-2298

**Comment by JeanChristopheMorinPerso** [comment]
@cbouss Can I do it in two stages, ie merge this PR as is and then create a new PR to update the version?

**Review by rafaelmartins — APPROVED** [review]

**Comment by cbouss** [comment]
> @cbouss Can I do it in two stages, ie merge this PR as is and then create a new PR to update the version?

That works too

**Review by cbouss — APPROVED** [review]

**Pull request merged**